pkg/cli/options.go

```diff
@@ -33,3 +34,67 @@ func ValidateOptions(opts apis.Validatable) func(cmd *cobra.Command, args []stri
 		return nil
 	}
 }
+
+type ListOptions struct {
+	Namespace     string
+	AllNamespaces bool
+}
+
+func (opts *ListOptions) Validate(ctx context.Context) *FieldError {
+	errs := &FieldError{}
+
+	if opts.Namespace == "" && !opts.AllNamespaces {
+		errs = errs.Also(ErrMissingOneOf("namespace", "all-namespaces"))
+	}
+	if opts.Namespace != "" && opts.AllNamespaces {
+		errs = errs.Also(ErrMultipleOneOf("namespace", "all-namespaces"))
+	}
+
+	return errs
+}
+
+type ResourceOptions struct {
+	Namespace string
+	Name      string
+}
+
+func (opts *ResourceOptions) Validate(ctx context.Context) *FieldError {
+	errs := &FieldError{}
+
+	if opts.Namespace == "" {
+		errs = errs.Also(ErrMissingField("namespace"))
+	}
+
+	if opts.Name == "" {
+		errs = errs.Also(ErrMissingField(opts.Name, "name"))
+	} else {
+		errs = errs.Also(validation.K8sName(opts.Name, "name"))
+	}
+
+	return errs
+}
+
+type DeleteOptions struct {
+	Namespace string
+	Names     []string
+	All       bool
+}
+
+func (opts *DeleteOptions) Validate(ctx context.Context) *FieldError {
+	errs := &FieldError{}
+
+	if opts.Namespace == "" {
+		errs = errs.Also(ErrMissingField("namespace"))
+	}
+
+	if opts.All && len(opts.Names) != 0 {
```

Comment:
maybe for later (or never :D), some kind of `xorValidation(condition, errorWhenAllFalse, errorWhenAllTrue)` function could be introduced (it's the second occurence of this kind in the PR)

Reply:
I've had the same thought. It gets even more fun when there are three or more values.